channels/rdpdr/client/rdpdr_main.c

```diff
@@ -973,45 +1171,46 @@ static void* rdpdr_virtual_channel_client_thread(void* arg)
 	return NULL;
 }
 
-static void rdpdr_virtual_channel_event_connected(rdpdrPlugin* rdpdr, LPVOID pData, UINT32 dataLength)
+static WIN32ERROR rdpdr_virtual_channel_event_connected(rdpdrPlugin* rdpdr, LPVOID pData, UINT32 dataLength)
 {
 	UINT32 status;
+	WIN32ERROR error;
 
 	status = rdpdr->channelEntryPoints.pVirtualChannelOpen(rdpdr->InitHandle,
 		&rdpdr->OpenHandle, rdpdr->channelDef.name, rdpdr_virtual_channel_open_event);
 
-	if (!rdpdr_add_open_handle_data(rdpdr->OpenHandle, rdpdr))
+	if ((error = rdpdr_add_open_handle_data(rdpdr->OpenHandle, rdpdr)))
 	{
-		WLog_ERR(TAG,  "%s: unable to register open handle", __FUNCTION__);
-		return;
+		WLog_ERR(TAG, "rdpdr_add_open_handle_data failed with error %lu!", error);
+		return error;
 	}
 
 	if (status != CHANNEL_RC_OK)
```

Comment:
This was already like this before the commit and maybe I'm wrong but shouldn't we do the check for status before rdpdr_add_open_handle_data is done?